### PR TITLE
Fallback to random port whenever already in use

### PR DESCRIPTION
**Pull request by japborst**
This PR adds the functionality for NepSnowplow to fall back to a random port whenever the initial one is in use. This is especially useful when debugging multiple devices at once, pointing each device to a different port. Alternatively, this comes in use when other applications already make use of the default port (3000).

### CODE REVIEW

File: server/Server.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                } else {
+                    console.log(err);
+                }
+            });
+    }
+
+    handleStartup(port) {
+        this.instance.emit("ready", port);
+        console.log("Listening for SnowPlow analytics on port " + port);
+        console.log("Please check you both of your devices are on the same network");
```
Comment (Stephan202) [inline server/Server.js]:
```suggestion
        console.log("Please check that both of your devices are on the same network");
```

File: server/Server.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        console.log(
+            "________________________________________________________________________________"
+        );
+        console.log("");
+    }
+
+    captureEvents() {
+        // Capturing every post events to this server
```
Comment (Stephan202) [inline server/Server.js]:
```suggestion
        // Capturing every post event sent to this server
```

File: server/Server.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    listen(port) {
+        let server = this;
+        this.instance
+            .listen(port, () => {
+                this.handleStartup(port);
+            })
+            .on("error", (err) => {
+                if (err.errno === "EADDRINUSE") {
+                    console.log("Port " + port + " in use, using random port");
+                    let randPort = Math.floor(1000 + Math.random() * 9000);
```
[the remark this answers is not in the document]
Comment (japborst) [reply server/Server.js]: You're right. Changed it into using `0`.

From what I read, OSX assigns the new port sequentially but doesn't check if that port is available or not. With the current implementation, it should detect if the next port is not available and try again until it finds an available port.

I'll test on an OSX device and write a Mocha test, as Travis tests on both OSX and Ubuntu.

File: test/main.js [hunk 4]
```diff
@@ -104,6 +108,31 @@ describe("NepSnowplow", function() {
                 });
         });
 
+        describe("second instance", function() {
```
[the remark this answers is not in the document]
Comment (larslockefeer) [reply test/main.js]: 👆

**Pull request merged**